examples/hybrid_jobs/6_QNSPSA_optimizer_with_embedded_simulator/qnspsa_with_embedded_simulator.ipynb

```diff
@@ -0,0 +1,1096 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e1c21f2b",
+   "metadata": {},
+   "source": [
+    "The step-wise execution time for the different optimizers in this QAOA problem:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "id": "8e0af8ec",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "GD takes 4.05 s per step\n",
+      "QNG takes 4.73 s per step\n",
+      "QNSPSA takes 0.42 s per step\n",
+      "SPSA takes 0.19 s per step\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(f'GD takes %.2f s per step' % (results[0]['GD_duration'] / repeats[0] / max_iter))\n",
+    "print(f'QNG takes %.2f s per step' % (results[1]['QNG_duration'] / repeats[1] / max_iter))\n",
+    "print(f'QNSPSA takes %.2f s per step' % (results[2]['QNSPSA_duration'] / repeats[2] / max_iter))\n",
+    "print(f'SPSA takes %.2f s per step' % (results[3]['SPSA_duration'] / repeats[3] / max_iter))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "63374753",
+   "metadata": {},
+   "source": [
+    "We can see the obvious advantage of the stochasitc optimizers (SPSA and QN-SPSA) in their execution time. Note that there are only 4 parameters in a depth-2 QAOA ansatz, yet it takes 10X more time for the gradient-based optimizers to finish a single optimization step, comparing to QN-SPSA (20X comparing to SPSA). The reason is that estimating the gradient with parameter shift rule requires to run 2$n_G$ circuits, where $n_G$ is the number of parametrized gates. The QAOA ansatz has limited parameters, but a rather large amount of parametrized gates applied. On the other hand, stochastic methods always execute a fixed number of circuits for each step. Combining the execution time observation with the optmization result, we can then clearly see the advantage of QN-SPSA in this use case."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "59a80aea",
+   "metadata": {},
+   "source": [
+    "## Significant advantage on large problems"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "86fd991e",
+   "metadata": {},
+   "source": [
+    "We would also like to confirm how well the optimizer works for a larger system. For that, we do the following stress test with a larger graph of 22 nodes and 44 edges. QN-SPSA needs 30 sec per step, and 3 hours overall to finish the optimization (single trace). On the other hand, it takes a GD optimizer more than 13 min to finish a single optimization step for a problem with this size. We can see that the analytical gradient estimation quickly become impractical when the system size gets larger. This is the regime where QN-SPSA demonstrates its full strength."
```

Comment:
```suggestion
    "The behavior and performance of different optimizers depend strongly on the problem instance and, in particular, the problem size.  As we have seen above, analytical gradient estimation, such as employed in GD and QNG quickly become impractical when the system size gets larger, due to the increasing number tasks and shots that are required to calculate the gradient. Stochastic optimizers, such as QN-SPSA are designed to overcome this challenge. Let's take a look how QN-SPSA performs as we go to larger system sizes. In the following cells, we simulate a larger QAOA graph with 22 nodes and 44 edges. Before you run the cell, note that QN-SPSA takes 30 seconds per step, and 3 hours overall to finish the algorithm"
```

Reply:
Why are we not running this on a bigger box? Or GPUs? This is the place to demonstrate the best performance!

Reply:
Provided a new example with 29 qubits on `lightning.gpu`.